src/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -584,11 +626,11 @@ public function create_item( $request ) {
 			);
 		}
 
-		// Do not allow comments to be created with a non-default type.
-		if ( ! empty( $request['type'] ) && 'comment' !== $request['type'] ) {
+		// Do not allow comments to be created with a non-core type.
+		if ( ! empty( $request['type'] ) && ! in_array( $request['type'], array( 'comment', 'note' ), true ) ) {
 			return new WP_Error(
 				'rest_invalid_comment_type',
-				__( 'Cannot create a comment with that type.' ),
+				__( 'Cannot create a comment with that type.', 'gutenberg' ),
```

Comment:
```suggestion
				__( 'Cannot create a comment with that type.' ),
```

Leftover text domain.